torch/_subclasses/meta_utils.py

```diff
@@ -89,19 +118,274 @@ def is_sparse_coo(t):
+            # https://github.com/pytorch/pytorch/issues/122089
+            crow_indices=self.describe_tensor(t.crow_indices(), recurse=False)
+            if recurse and t.layout in {torch.sparse_csr, torch.sparse_bsr}
+            else None,
+            col_indices=self.describe_tensor(t.col_indices(), recurse=False)
+            if recurse and t.layout in {torch.sparse_csr, torch.sparse_bsr}
+            else None,
+            ccol_indices=self.describe_tensor(t.ccol_indices(), recurse=False)
+            if recurse and t.layout in {torch.sparse_csc, torch.sparse_bsc}
+            else None,
+            row_indices=self.describe_tensor(t.row_indices(), recurse=False)
+            if recurse and t.layout in {torch.sparse_csc, torch.sparse_bsc}
+            else None,
+            values=self.describe_tensor(t.values(), recurse=False)
+            if recurse and is_sparse_compressed(t)
+            else None,
+            grad=self.describe_tensor(safe_grad(t))
+            if safe_grad(t) is not None
+            else None,
+            creation_meta=torch._C._autograd._get_creation_meta(t)
+            if t._is_view()
+            else None,
+            unwrapped=self.describe_tensor(get_unwrapped(t))
+            if is_batchedtensor_v or is_gradtrackingtensor_v
+            else None,
+            level=maybe_get_level(t)
+            if is_batchedtensor_v or is_gradtrackingtensor_v
+            else None,
+            bdim=maybe_get_bdim(t) if is_batchedtensor_v else None,
+            base=self.describe_tensor(t._base)
+            if recurse and t._is_view() and t._base is not None
+            else None,
+            fake_mode=torch._subclasses.fake_tensor.maybe_get_fake_mode(t),
+            view_func=t._view_func_unsafe,
+            attrs=attrs,
+            ctx=ctx,
+            type=type_v,
+        )
+
+
+@dataclass(frozen=True)
+class MetaStorageDesc:
+    id: MetaStorageId
+    size: int
+
+
+@dataclass(frozen=True)
+class MetaTensorDesc:
+    id: MetaTensorId
+    is_inference: bool
+    is_leaf: bool
+    requires_grad: bool
+    ndim: int
+    dtype: torch.dtype
+    is_sparse: bool
+    is_mkldnn: bool
+    is_functorch_wrapped: bool
+    is_batchedtensor: bool
+    is_gradtrackingtensor: bool
+    is_view: bool
+    is_nested: bool
+    is_traceable_wrapper_subclass: bool
+    is_conj: bool
+    is_neg: bool
+    layout: torch.layout
+    # NB: Sometimes, size, stride and storage_offset contain SymInt, in which
+    # case this is NOT serializable.  That only happens when you're
+    # re-fakeifying a fake tensor with an existing ShapeEnv... maybe we
+    # can get rid of this use case entirely
+    # NB: size could potentially be None as you can override it and make it
+    # throw an error, but we don't currently have any subclasses that do this
+    # except C++ nested tensor but we're going to have nested int to make this
+    # defined on NJT
+    size: Tuple[int, ...]
```

Comment:
No action necessary, but I'm just trying to think through how I'd update the recursive metafication logic once I add tensor as a field to nested int, and it seems like we would need to have extend this to support SymInts that can exist eager, e.g. introduce some kind data class for eager SymInts which would then recursively contain another MetaTensorDesc.